### ISSUE TITLE
horse25519 is not a library

### ISSUE THREAD

**Issue by Yawning** [MystenLabs/ed25519-unsafe-libs#4]
The repository is intended to provide a standalone executable that does ed25519 vanity keypair generation.  It includes a copy of djb's ref10 ed25519 implementation lifted from supercop to avoid pulling in another dependency.

While it does use the API in an odd way, this is intentional as it already is doing something extremely exotic and unusual with respect to key generation.

https://github.com/Yawning/horse25519/blob/master/Makefile#L57

Edit: Corrected why the included sign API is somewhat wacky.

**Comment by kchalkias** [comment]
Thanks for the update, I'll push a PR to remove it from the official list and explain that carrying the vulnerable public ref10 api is not meant to be used externally. In theory though you can still have a copy of that lib but make that function private/protected so the vulnerable api is not exposed in the first place.

**Comment by kchalkias** [comment]
@Yawning a possible clear solution to avoid any accidental public exposure of the horse25519 repo is to completely remove the public signing api (and any other non-used api) from the copy pasted ref10 dep.

**Comment by Yawning** [comment]
@kchalkias I don't disagree in principle, and acknowledge that people do tend to copy/paste and or cargo-cult a good amount of code.

That said, I would hope that no one will copy/paste things from the innards of a long-abandoned proof of concept, in a repository that I archived years ago.  While I may still be open to fixing this since I can probably just remove the sign api, there has to be a way for software authors to disown responsibility for maintaining code (short of deleting the repository, since the concept that I was trying to demonstrate is still useful).

**Comment by Yawning** [comment]
Whowa, didn't think my comment in my repository would close this, sorry about that.

As the PR notes, I went and changed my modified crypto_sign to do the scalar-basepoint multiply, so even if someone copy-pastes it, it will be "safe".  It is somewhat unlikely that anyone would use a crypto_sign that takes an expanded form private key, but, I've seen code I wrote crop up in weirder places.

As a side note, thanks for the EdDSA verification semantics paper from a while back, it came in handy for a separate project.
